code/game/objects/items/devices/uplinks.dm

```diff
@@ -47,14 +48,17 @@ GLOBAL_LIST_EMPTY(world_uplinks)
 /**
   * Build the item lists for use with the UI
   *
-  * Generates a list of items for use in the UI, based on job and other parameters
+  * Generates a list of items for use in the UI, based on job, species and other parameters
   *
   * Arguments:
   * * user - User to check
   */
 /obj/item/uplink/proc/generate_item_lists(mob/user)
 	if(!job)
 		job = user.mind.assigned_role
+	var/species
```

Comment:
Sorry, I didn't see that this is called every time you open the UI. You can remove this and keep the class-level var.
```suggestion
```

Reply:
I /think/ this is fixed? Let me know if it isn't.